src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.Web/WeatherApiClient.cs

```diff
@@ -2,9 +2,24 @@ namespace AspireStarterApplication._1.Web;
 
 public class WeatherApiClient(HttpClient httpClient)
 {
-    public async Task<WeatherForecast[]> GetWeatherAsync()
+    public async Task<WeatherForecast[]> GetWeatherAsync(int maxItems = 10, CancellationToken cancellationToken = default)
     {
-        return await httpClient.GetFromJsonAsync<WeatherForecast[]>("/weatherforecast") ?? [];
+        List<WeatherForecast>? forecasts = null;
+
+        await foreach (var forecast in httpClient.GetFromJsonAsAsyncEnumerable<WeatherForecast>("/weatherforecast", cancellationToken))
```

Comment:
Shouldn't the maxItems also be flown to the backend?

Reply:
This is about ensuring the client isn't making unbounded calls to a remote endpoint. Whether the server supports paging/limits or not isn't as important.

Reply:
Isn't the remote call from `.Web` => `.Api` still unbounded? This is just changing `.Web` to not consume all the results.

Reply:
`.Web` will stop when it gets to 10 items (and any default timeout, etc.), as opposed to just reading and creating as many items as the backend returns. That was the purpose of the change.

Reply:
But isn't it a problem for the backend/`.Api`? To get an unbounded request? Or are we saying because it controls how many it returns, it isn't a problem?

Reply:
The backend isn't written to produce unlimited items so there's no such thing as an unbounded request (if I understand your point).

Reply:
But as soon as you change it to return the items from somewhere, like a database, don't you have the problem?

Reply:
We can't pre-empt all possible issues in the template code for what an API might have its implementation changed to, but the client doesn't have knowledge of the server implementation so doing a check there is prudent.

Reply:
OK, if it isn't "standard practice" to put a limit on the API call, then I guess this is fine.